det3pa/med3pa/experiment.py

```diff
@@ -0,0 +1,353 @@
+            apc_grid_params (dict, optional): Grid search parameters for optimizing the APC model, by default None.
+            apc_cv (int, optional): Number of cross-validation folds for optimizing the APC model, by default None.
+            samples_ratio_min (int, optional): Minimum sample ratio, by default 0.
+            samples_ratio_max (int, optional): Maximum sample ratio, by default 50.
+            samples_ratio_step (int, optional): Step size for sample ratio, by default 5.
+            med3pa_metrics (list of str, optional): List of metrics to calculate, by default ['Auc', 'Accuracy', 'BalancedAccuracy'].
+            evaluate_models (bool, optional): Whether to evaluate the models, by default False.
+            models_metrics (list of str, optional): List of metrics for model evaluation, by default ['MSE', 'RMSE'].
+
+        Returns:
+            Med3paResults: the results of the MED3PA experiment.
+        """
+        # retrieve the dataset based on the set type
+        if set == 'reference':
+            dataset = datasets_manager.get_dataset_by_type(dataset_type="reference", return_instance=True)
+        elif set == 'testing':
+            dataset = datasets_manager.get_dataset_by_type(dataset_type="testing", return_instance=True)
+        else:
+            raise ValueError("The set must be either the reference set or the testing set")
+
+        # retrieve different dataset components to calculate the metrics
+        x = dataset.get_features()
+        y_true = dataset.get_true_labels()
+        features = datasets_manager.get_column_labels()
+
+        # Initialize base model and predict probabilities if not provided
+        if base_model_manager is None and predicted_probabilities is None:
+            raise ValueError("Either the base model or the predicted probabilities should be provided!")
+        
+        if predicted_probabilities is None:
+            base_model = base_model_manager.get_instance()
+            predicted_probabilities = base_model.predict(x, True)
+
+        dataset.set_pseudo_labels(predicted_probabilities)
+
+        # Calculate uncertainty values
+        uncertainty_calc = UncertaintyCalculator(uncertainty_metric)
+        uncertainty_values = uncertainty_calc.calculate_uncertainty(x, predicted_probabilities, y_true)
+
+        # set predicted labels
+        dataset.set_pseudo_probs_labels(predicted_probabilities, 0.5)
+        
+        if evaluate_models:
+            x_train, x_test, uncertainty_train, uncertainty_test = train_test_split(x, uncertainty_values, test_size=0.1, random_state=42)
+        else:
+            x_train = x
+            uncertainty_train = uncertainty_values
+
+        # Create and train IPCModel
+        IPC_model = IPCModel(ipc_type, ipc_params)
+        
+        IPC_model.train(x_train, uncertainty_train)
+        print("IPC Model training completed.")
```

Comment:
Juste une suggestion pour éviter des lignes collées

# Create the IPCModel
IPC_model = IPCModel(ipc_type, ipc_params)

# Train the IPCModel
IPC_model.train(x_train, uncertainty_train)

print("IPC Model training completed.")